src/rt/test/func/promise/promise.cc

```diff
@@ -16,15 +16,33 @@ void promise_test()
     Promise<int>::fulfill(std::move(wp), 42);
   });
 
-  rp.then(
-    [](int val) { Systematic::cout() << val << std::endl; },
-    []() { Systematic::cout() << "Error\n"; });
-  rp.then(
-    [](int val) { Systematic::cout() << val << std::endl; },
-    []() { Systematic::cout() << "Error\n"; });
-  rp2.then(
-    [](int val) { Systematic::cout() << val << std::endl; },
-    []() { Systematic::cout() << "Error\n"; });
+  rp.then([](std::variant<int, Promise<int>::PromiseErr> val) {
+    if (std::holds_alternative<int>(val))
+    {
+      auto v = std::get<int>(std::move(val));
```

Comment:
It would be useful to know which bits are expected and not expected to be reached in this example.  Could you put `abort` on the cases you don't expect to be called.

Reply:
Good idea. The promise_no_writer case will become cleaner this way as well. I'm adding them now.